### PR TITLE
Added handling of gameEnd firebase message

### LINKED ISSUE #30: Handling fcm message for the end of the game.
The end of the game message will include winners list and chips won by each player.

### PR DESCRIPTION
**Pull request by Kwasow**
This PR adds handling of a new firebase message for ending the game as well as showing the winner and passing the winnings onto them.

I also worked on how the FCM message names are stored.

Closes #30

### CODE REVIEW

File: app/src/main/java/com/pokerio/app/PokerioFirebaseMessagingService.kt [hunk 1]
```diff
@@ -114,15 +125,20 @@ class PokerioFirebaseMessagingService : FirebaseMessagingService() {
...
         @OptIn(ExperimentalSerializationApi::class)
         fun newCards(data: Map<String, String>) {
             PokerioLogger.debug("Received newCards FCM message")
             GameState.newCards(Json.decodeFromString(data["cards"]!!))
         }
+
+        @OptIn(ExperimentalSerializationApi::class)
+        fun endGame(data: Map<String, String>) {
+            PokerioLogger.debug("Received gameEnd FCM message")
+            println(data["winners"])
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/PokerioFirebaseMessagingService.kt]: Leftover debug?

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 2]
```diff
@@ -389,21 +389,38 @@ object GameState {
...
         winningsPool += player.bet
 
         newActionCallbacks.forEach { it.value(player) }
     }
 
-    fun handleActionWon(playerHash: String) {
-        val isThisPlayer = sha256(thisPlayer.playerID) == playerHash
+    fun handleActionWon(winners: List<String>, amount: Int) {
+        winners.forEach { playerHash ->
+            val isThisPlayer = sha256(thisPlayer.playerID) == playerHash
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]: Why is this a specyfic case? We don't use anywhere else, so why won't we look just for player with expected ID?

**Review by sormys — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/pokerio/app/PokerioFirebaseMessagingService.kt (hunk 1 above)
Comment (Kwasow) [reply app/src/main/java/com/pokerio/app/PokerioFirebaseMessagingService.kt]: Yep, removed it

File: app/src/main/java/com/pokerio/app/utils/GameState.kt (hunk 2 above)
Comment (Kwasow) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt]: We do use it - remember that all we have hashes of every player apart from our own - this is how we recognize our own player (although maybe we should change it). Therefor we need this to know if we want to take our player or if we want to search for the player using the standard (non `sha256()`) search.

Comment (Kwasow) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt]: Although I will put the search into a separate function, as we do it multiple times in different functions

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 3]
```diff
@@ -560,6 +550,18 @@ object GameState {
     fun getMaxBet(): Int {
         return players.maxOf { it.bet }
     }
+
+    private fun getPlayer(id: String): Player {
+        val isThisPlayer = isThisPlayersId(id)
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt]: This variable is unnecessary. You can just use isThisPlayersId(id) in `if`.

Comment (Kwasow) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt]: Good catch!

**Review by sormys — APPROVED** [review]

**Pull request merged**